google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/DefaultAppEngineProjectService.java

```diff
@@ -228,19 +230,21 @@ public AppEngineStandardRuntime getAppEngineStandardDeclaredRuntime(
 
   private Optional<String> getValueFromAppYaml(@NotNull String appYamlPathString,
       @NotNull String key) {
-    Path appYamlPath;
+    Yaml yamlParser = new Yaml();
     try {
-      appYamlPath = Paths.get(appYamlPathString);
-    } catch (InvalidPathException ipe) {
-      return Optional.empty();
-    }
+      Object parseResult = yamlParser.load(
+          Files.newBufferedReader(Paths.get(appYamlPathString), Charset.defaultCharset()));
 
-    try {
-      return Files.readAllLines(appYamlPath).stream()
-          .filter(line -> line.startsWith(key))
-          .map(line -> line.substring(key.length()).trim())
-          .findFirst();
-    } catch (IOException ioe) {
+      if (!(parseResult instanceof Map)) {
+        return Optional.empty();
+      }
+
+      // It's possible to get rid of this unchecked cast using a loadAs(file,
+      // AppEngineYamlWebApp.class) sort of approach.
+      Map<String, String> yamlMap = (Map<String, String>) parseResult;
+
+      return yamlMap.containsKey(key) ? Optional.of(yamlMap.get(key)) : Optional.empty();
+    } catch (InvalidPathException | IOException ioe) {
```

Comment:
don't we want to differentiate from a missing file or other IO exception from a missing yaml element?